mautic/mautic
# Amazon SES bounce&complaint webhook

Please answer the following questions. 

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | yes | 
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/92
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

**Note that all new features should have a related user and/or developer documentation PR in their respective repositories.** 

### Required
#### Description:
Extend the Amazon mail transport with ability to receive bounce and complaint notifications from Amazon SES/SNS.

#### Steps to test this PR:
1. Enable the Amazon transport
1. Create a lead with a non-existing e-mail address
1. Configure Amazon SES to send bounce&complaints into a Amazon SNS topic
1. Create a subscriber in Amazon SNS pointing to http://your-mautic/mailer/amazon/callback
1. Make sure the  subscriber changed from pending to confirmed (the transport will automatically call the subscriber url)
1. Send an e-mail to the non-existing e-mail address
1. Go to /s/contacts and check that the lead is marked as bounce

I'll write some documentation how to setup when you agree with the implementation
